### PR TITLE
Use `CommonDatabase` interface for sqlite3

### LINKED ISSUE #263: Use conditional imports OR move FFI bits out of core sdk
When compiling for web clients, we cannot use Dart libraries that use `dart:ffi` as a dependency; therefore, we should exclude it from builds (e.g., [tiki-sdk-js](https://github.com/tiki/tiki-sdk-js)).

### Option 1: Conditional Imports

In Dart 2, support for custom flags was removed. Support for flags for dart internal libraries remains an **undocumented** feature.  dart-lang/sdk#34262. 

#### Cons

- More dependencies of this project that may be experimental and untested.
- There is no indication of whether the conditional import pattern will work in Dart 3.

### Option 2: Remove `dart:ffi` specific bits from the core SDK

Another option is to remove `dart:ffi` based implementations from the core SDK altogether.

#### Cons

- Requires update to flutter SDK because a lot of it is _for_ flutter, I think.
- A lot more work

### Option 3: Refactor import statements

Refactoring the imports to be less spaghetti-like could make it easy to outright _avoid_ `dart:ffi` & friends.

#### Cons:

- We'll have to re-implement key parts of the interface so that we only tap into the classes we want
- We may find that we still need to do conditional imports or move FFI bits out

### PR DESCRIPTION
**Pull request by jcockhren**
This PR updates the internal interfaces to use the `CommonDatabase` class to ease web platform builds. Leverages info from: https://pub.dev/packages/sqlite3#wasm-web-support

Updated all references to the `Database` class to use `CommonDatabase.`

FYI: This PR is marked for review; however, I'd like to postpone the merge until we've done more testing with sqlite3 WASM on the JS side.

closes #263

### CODE REVIEW

**Comment by jcockhren** [comment]
Superseded by PR #265

**Comment by ricardobrg** [comment]
@jcockhren I have changed the status to draft because you want to merge it later. Please update the status when you think it is good to go.

File: test/ownership/ownership_test.dart [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 import 'dart:typed_data';
 
+import 'package:sqlite3/common.dart';
 import 'package:sqlite3/sqlite3.dart';
```
Comment (ricardobrg) [inline test/ownership/ownership_test.dart:9]: Do we need both imports here?

Comment (jcockhren) [reply test/ownership/ownership_test.dart]: @ricardobrg Something that your review points out is that in order downstream, libraries will likely have to import both '`sqlite3/common.dart` and `sqlite3/sqlite3.dart`. The former provides us `CommonDatabase`, and the later provides use the `sqlite` property.

**Comment by ricardobrg** [comment]
Adding @mike-audi to the review since he is working in Dart SDK 2.0

File: lib/helpers/sdk_builder.dart [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import '../node/key/key_storage.dart';
+import '../tiki_sdk.dart' show TikiSdk;
+
+abstract class SdkBuilder {
```
Comment (mike-audi) [inline lib/helpers/sdk_builder.dart:4]: FYI: In API 2.0 the builder is gone/moved into a static init in TikiSdk.dart

**Review by mike-audi — APPROVED** [review]

**Comment by mike-audi** [comment]
FYI @jcockhren @ricardobrg, I'm going to merge this into my 2.0 branch so I can test it with my updated code before merging back into main.

**Comment by mike-audi** [comment]
BTW @jcockhren feel free to stay on this branch if you'd prefer after I merge.

**Pull request merged**